src/modules/fundingManager/bondingCurve/FM_BC_Bancor_Redeeming_VirtualSupply_v1.sol

```diff
@@ -541,10 +541,23 @@ contract FM_BC_Bancor_Redeeming_VirtualSupply_v1 is
         internal
         override(VirtualIssuanceSupplyBase_v1)
     {
+        uint minRequiredSupply;
+        uint decimalDifference;
+
+        if (issuanceTokenDecimals <= 18) {
+            // If issuance token has 18 or fewer decimals. Ensure that decimal difference is at least 1
+            // to end up with minRequiredSupply of 10
+            decimalDifference =
+                18 - issuanceTokenDecimals == 0 ? 1 : 18 - issuanceTokenDecimals;
+        } else {
+            // If issuance token has more than 18 decimals
+            decimalDifference = issuanceTokenDecimals - 18;
+        }
+        minRequiredSupply = 10 ** decimalDifference;
         // Check if virtual supply is big enough to ensure compatibility with relative issuance
         // token decimal and conversion to 18 decimals done in FM_BC_Tools._convertAmountToRequiredDecimal()
         // so it will not result in a round down 0 value
-        if (_virtualSupply < 10 ** (issuanceTokenDecimals - 18)) {
+        if (_virtualSupply < minRequiredSupply) {
```

Comment:
Logic is correct from what I checked. Just some gas stuff.

Caching the `issuanceTokenDecimals` saves gas, as we otherwise access contract state 4 times. We also declare `minRequiredSupply` above, set it later and then just use it once, so it's cheaper to just use that directly, saves an `MSTORE` and an `MLOAD`.

```suggestion
        uint decimalDifference;
        uint decimals = issuanceTokenDecimals;

        if (decimals <= 18) {
            // If issuance token has 18 or fewer decimals. Ensure that decimal difference is at least 1
            // to end up with minRequiredSupply of 10
            decimalDifference =
                18 - decimals == 0 ? 1 : 18 - decimals;
        } else {
            // If issuance token has more than 18 decimals
            decimalDifference = decimals - 18;
        }
        
        // Check if virtual supply is big enough to ensure compatibility with relative issuance
        // token decimal and conversion to 18 decimals done in FM_BC_Tools._convertAmountToRequiredDecimal()
        // so it will not result in a round down 0 value
        if (_virtualSupply < 10 ** decimalDifference) {
```